core/domain/exp_jobs_one_off.py

```diff
@@ -678,10 +682,45 @@ def map(file_model):
                             content, mimetype='image/%s' % filetype)
                         yield (FILE_COPIED, 1)
 
-
     @staticmethod
     def reduce(status, values):
         if status == FILE_COPIED:
             yield (status, len(values))
         else:
             yield (status, values)
+
+
+class CreateVersionsOfImageJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for creating compressed versions of the images
+    of the exploration in the GCS.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(exp_model):
+        exploration = exp_services.get_exploration_by_id(exp_model.id)
+        current_version = exploration.version
+        filenames = []
+        for version in range(current_version):
+            exploration = exp_services.get_exploration_by_id(
```

Comment:
Oh, one more thing, sorry. Can you do a get_multi() here instead?

Reply:
I am not sure. I'll have to look into it. But this is approach works fine.

Reply:
There is get_multi_versions function in VersionedModel. For that we need to send the list of the versions required and it gives the list of the models in return. My approach is similar, anyhow we'll have to have one iteration which will iterate over explorations.
Shall I change it or keep it as it is?

Reply:
Change it. get_multi makes one datastore call, compared to get() which makes many datastore calls.